### PR TITLE
SPARK-7265 Improving documentation for Spark SQL Hive support

### PR DESCRIPTION
**Pull request by JihongMA**
Please review this pull request.

### CODE REVIEW

File: docs/running-on-yarn.md [hunk 1]
```diff
@@ -305,3 +305,4 @@ If you need a reference to the proper location to put log files in the YARN so t
...
 - The `--files` and `--archives` options support specifying file names with the # similar to Hadoop. For example you can specify: `--files localtest.txt#appSees.txt` and this will upload the file you have locally named localtest.txt into HDFS but this will be linked to by the name `appSees.txt`, and your application should use the name as `appSees.txt` to reference it when running on YARN.
 - The `--jars` option allows the `SparkContext.addJar` function to work if you are using it with local files and running in `yarn-cluster` mode. It does not need to be used if you are using it with HDFS, HTTP, HTTPS, or FTP files.
+- In order to make PySpark work on Yarn, please build spark with Jave 6.
```
Comment (rxin) [inline docs/running-on-yarn.md]: actually we should really fix this ...

Comment (srowen) [inline docs/running-on-yarn.md]: Jave -> Java; Yarn -> YARN. But this is probably out of date soon anyway as Java 7 will be required.

File: docs/sql-programming-guide.md [hunk 2]
```diff
@@ -1253,7 +1253,12 @@ This command builds a new assembly jar that includes Hive. Note that this Hive a
 on all of the worker nodes, as they will need access to the Hive serialization and deserialization libraries
 (SerDes) in order to access data stored in Hive.
 
-Configuration of Hive is done by placing your `hive-site.xml` file in `conf/`.
+Configuration of Hive is done by placing your `hive-site.xml` file in `conf/`. Please note when running
```
Comment (srowen) [inline docs/sql-programming-guide.md]: I don't think this is true; it just has to be on the classpath?

Comment (JihongMA) [inline docs/sql-programming-guide.md]: the datanucleus jars are added to spark classpath through compute-classpath.sh, but it is not actually added in yarn container. there was JIRA opened to track this SPARK-2624, but the fix is reverted in Dec last year. 

http://mail-archives.apache.org/mod_mbox/incubator-spark-commits/201412.mbox/%3Ceea141c4011644c6b91a8a202a131513@git.apache.org%3E

File: docs/running-on-yarn.md (hunk 1 above)
Comment (JihongMA) [inline docs/running-on-yarn.md]: there are couple JIRAs opened to enable PySpark support with Java 7, are we expecting them to be fixed in Spark 1.4?

File: docs/sql-programming-guide.md (hunk 2 above)
Comment (srowen) [inline docs/sql-programming-guide.md]: I meant `hive-site.xml` specifically? 
The Java 6 comment is out of date or will be soon I believe.

Comment (JihongMA) [inline docs/sql-programming-guide.md]: do you mean adding hive-site.xml through spark.driver.extraClassPath? when running on yarn-cluster mode, the AM node which runs the driver program can be any node on the cluster, and that node may not have Spark client installed.

Comment (srowen) [inline docs/sql-programming-guide.md]: You can disregard my comment; I think I'm mixed up here and so the current comment should stand. Last nit: Yarn -> YARN in the line below. I'd also remove the Java 6 comment as this is obsolete already.

File: docs/sql-programming-guide.md [hunk 3]
```diff
@@ -1258,7 +1258,12 @@ This command builds a new assembly jar that includes Hive. Note that this Hive a
 on all of the worker nodes, as they will need access to the Hive serialization and deserialization libraries
 (SerDes) in order to access data stored in Hive.
 
-Configuration of Hive is done by placing your `hive-site.xml` file in `conf/`.
+Configuration of Hive is done by placing your `hive-site.xml` file in `conf/`. Please note when running 
+the query on YARN cluster (`yarn-cluster` mode),the `datanucleus` jars under `lib_managed/jars` directory
```
Comment (srowen) [inline docs/sql-programming-guide.md]: comma needs a space after, add "the" before spark-submit and put it in `code font`.

Comment (srowen) [inline docs/sql-programming-guide.md]: Fixed the grammar myself on merge

**Pull request closed without merging**